pkg/reconciler/taskrun/config.go

```diff
@@ -0,0 +1,391 @@
+		AdditionalInstanceTags: make(map[string]string),
+	}
+
+	// Parse local platforms
+	localPlatforms, err := parsePlatformList(data[LocalPlatforms], "local")
+	if err != nil {
+		return nil, err
+	}
+	config.LocalPlatforms = localPlatforms
+
+	// Parse dynamic platforms
+	dynamicPlatforms, err := parsePlatformList(data[DynamicPlatforms], "dynamic")
+	if err != nil {
+		return nil, err
+	}
+	for _, platform := range dynamicPlatforms {
+		platformConfigName := strings.ReplaceAll(platform, "/", "-")
+		prefix := "dynamic." + platformConfigName + "."
+
+		// Parse and validate dynamic platform configuration
+		dynamicConfig := DynamicPlatformConfig{}
+
+		// Type field (required)
+		if typeName := data[prefix+"type"]; typeName != "" {
+			dynamicConfig.Type = typeName
+		} else {
+			return nil, fmt.Errorf("dynamic platform '%s': type field is required", platform)
+		}
+
+		// Max instances (required)
+		if maxInstancesStr := data[prefix+"max-instances"]; maxInstancesStr != "" {
+			maxInstances, err := validateMaxInstances(maxInstancesStr)
+			if err != nil {
+				return nil, fmt.Errorf("dynamic platform '%s': invalid max-instances '%s': %w", platform, maxInstancesStr, err)
+			}
+			dynamicConfig.MaxInstances = maxInstances
+		} else {
+			return nil, fmt.Errorf("dynamic platform '%s': max-instances field is required", platform)
+		}
+
+		// Instance tag (optional)
+		if instanceTag := data[prefix+"instance-tag"]; instanceTag != "" {
+			if err := validateDynamicInstanceTag(platformConfigName, instanceTag); err != nil {
+				return nil, fmt.Errorf("dynamic platform '%s': invalid instance-tag '%s': %w", platform, instanceTag, err)
+			}
+			dynamicConfig.InstanceTag = instanceTag
+		}
+
+		// Allocation timeout (optional)
+		if timeoutStr := data[prefix+"allocation-timeout"]; timeoutStr != "" {
+			timeout, err := validateMaxAllocationTimeout(timeoutStr)
+			if err != nil {
+				return nil, fmt.Errorf("dynamic platform '%s': invalid allocation-timeout '%s': %w", platform, timeoutStr, err)
+			}
+			dynamicConfig.AllocationTimeout = int64(timeout)
+		} else {
+			// Default to 10 minutes if not specified
+			dynamicConfig.AllocationTimeout = int64(defaultAllocationTimeout)
+		}
+
+		// SSH secret (required)
+		if sshSecret := data[prefix+"ssh-secret"]; sshSecret != "" {
+			// For AWS platforms, just validate that there is a value
+			if dynamicConfig.Type == "aws" {
+				// AWS platforms: ensure the field has a non-empty value
+				if strings.TrimSpace(sshSecret) == "" {
+					return nil, fmt.Errorf("dynamic platform '%s': ssh-secret cannot be empty for AWS platforms", platform)
+				}
+				dynamicConfig.SSHSecret = sshSecret
+			} else {
+				// IBM platforms: validate using validateIBMHostSecret
+				if err := validateIBMHostSecret(platform, sshSecret); err != nil {
+					return nil, fmt.Errorf("dynamic platform '%s': invalid ssh-secret '%s': %w", platform, sshSecret, err)
+				}
+				dynamicConfig.SSHSecret = sshSecret
+			}
```

Comment:
We should not assume that if `dynamicConfig.Type` is not `aws` then it's IBM. Let's make sure the value meets our expectations.

```suggestion
			switch dynamicConfig.Type {
			case "aws":
				// AWS platforms: ensure the field has a non-empty value
				if strings.TrimSpace(sshSecret) == "" {
					return nil, fmt.Errorf("dynamic platform '%s': ssh-secret cannot be empty for AWS platforms", platform)
				}
				dynamicConfig.SSHSecret = sshSecret
			case "ibm":
				// IBM platforms: validate using validateIBMHostSecret
				if err := validateIBMHostSecret(platform, sshSecret); err != nil {
					return nil, fmt.Errorf("dynamic platform '%s': invalid ssh-secret '%s': %w", platform, sshSecret, err)
				}
				dynamicConfig.SSHSecret = sshSecret
			default:
			       return nil, fmt.Errorf("invalid type: expect 'ibm' or 'aws', got '%s', dynamicConfig.Type)
			}
```

Reply:
Added to changes commit